File: backend/Data/Entities/Ticket.cs
namespace backend.Data.Entities;

public class Ticket
{
    public int Id { get; set; }
    public string Code { get; set; } = "";
    public string Subject { get; set; } = "";
    public int RequesterId { get; set; }
    public int RoomId { get; set; }
    public int DeviceId { get; set; }
    public string Category { get; set; } = "";
    public string Priority { get; set; } = "";
    public string Status { get; set; } = "";
    public string Channel { get; set; } = "";
    public int AssigneeId { get; set; }
    public string Sla { get; set; } = "";
    public string CreatedAt { get; set; } = "";
    public string UpdatedAt { get; set; } = "";

    // Navigation properties
    public AppUser? Requester { get; set; }
    public Room? Room { get; set; }
    public Device? Device { get; set; }
    public AppUser? Assignee { get; set; }
}

File: frontend/src/types/ticket.ts
import type { KpiDto, PointDto } from './common'

export type TicketPriority = 'Khẩn cấp' | 'Cao' | 'Trung bình' | 'Thấp'

export type TicketStatus = 'Mới' | 'Đang xử lý' | 'Chờ phản hồi' | 'Hoàn thành' | 'Đóng'

export type TicketChannel = 'QR' | 'Email' | 'Điện thoại' | 'Trực tiếp' | 'Hệ thống'

export type TicketItem = {
  id: string
  code: string
  subject: string
  requester: string
  room: string
  device: string
  category: string
  priority: TicketPriority
  status: TicketStatus
  channel: TicketChannel
  assignee: string
  sla: string
  createdAt: string
  updatedAt: string
}

export interface TicketManagementResponse {
  overview: KpiDto[]
  statusData: PointDto[]
  priorityData: PointDto[]
  channelData: PointDto[]
  slaTrendData: PointDto[]
  ageBuckets: PointDto[]
  items: TicketItem[]
  statuses: string[]
  priorities: string[]
  channels: string[]
}
